addons/account/report/account_invoice_report.py

```diff
@@ -56,131 +56,137 @@ def _compute_amounts_in_user_currency(self):
+                    move.name,
+                    move.state,
+                    move.type,
+                    ROUND(move.amount_residual * currency_table.rate, currency_table.precision) AS amount_residual,
+                    ROUND(move.amount_total * currency_table.rate, currency_table.precision)    AS amount_total,
+                    move.partner_id,
+                    move.user_id,
+                    move.fiscal_position_id,
+                    move.invoice_payment_state,
+                    move.invoice_date,
+                    move.invoice_date_due,
+                    move.invoice_payment_term_id,
+                    move.invoice_partner_bank_id,
+                    uom_template.name                                                       AS uom_template_name,
+                    template.categ_id,
+                    SUM(NULLIF(
+                        (CASE WHEN move.type IN ('out_refund', 'in_invoice') THEN -1 ELSE 1 END * line.quantity)
+                        / (COALESCE(uom_line.factor, 1) * COALESCE(uom_template.factor, 1)), 1.0
+                    ))                                                                      AS quantity,
+                    SUM((
+                        CASE WHEN move.type IN ('out_refund', 'in_invoice')
+                        THEN -1 ELSE 1 END * line.price_subtotal
+                    ))                                                                      AS price_subtotal,
+                    SUM(line.price_subtotal / NULLIF(
+                        (CASE WHEN move.type IN ('out_refund', 'in_invoice') THEN -1 ELSE 1 END * line.quantity)
+                        / (COALESCE(uom_line.factor, 1) * COALESCE(uom_template.factor, 1)), 1.0
+                    ))                                                                      AS price_average,
+                    COALESCE(partner.country_id, commercial_partner.country_id)             AS country_id
+                FROM account_move_line line
+                    LEFT JOIN res_partner partner ON partner.id = line.partner_id
+                    LEFT JOIN product_product product ON product.id = line.product_id
+                    LEFT JOIN account_account account ON account.id = line.account_id
+                    LEFT JOIN account_account_type user_type ON user_type.id = account.user_type_id
+                    LEFT JOIN product_template template ON template.id = product.product_tmpl_id
+                    LEFT JOIN uom_uom uom_line ON uom_line.id = line.product_uom_id
+                    LEFT JOIN uom_uom uom_template ON uom_template.id = template.uom_id
+                    INNER JOIN account_move move ON move.id = line.move_id
+                    LEFT JOIN res_partner commercial_partner ON commercial_partner.id = move.commercial_partner_id
+                    LEFT JOIN ''' + currency_table + ''' ON currency_table.company_id = line.company_id
```

Comment:
why is this currency_table only used for the amount_residual and amount_total columns?

You should use it for any monetary value